drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c

```diff
@@ -868,4 +888,110 @@ int wifi_nrf_wpa_set_supp_port(void *if_priv, int authorized, char *bssid)
 
 	return wifi_nrf_fmac_chg_sta(rpu_ctx_zep->rpu_ctx, vif_ctx_zep->vif_idx, &chg_sta_info);
 }
+
+int wifi_nrf_wpa_supp_signal_poll(void *if_priv, struct wpa_signal_info *si, unsigned char *bssid)
+{
+	struct wifi_nrf_vif_ctx_zep *vif_ctx_zep = NULL;
+	struct wifi_nrf_ctx_zep *rpu_ctx_zep = NULL;
+	struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx = NULL;
+	enum wifi_nrf_status ret = WIFI_NRF_STATUS_FAIL;
+
+	if (!if_priv || !si || !bssid) {
+		LOG_ERR("%s: Invalid params\n", __func__);
+		goto out;
+	}
+
+	vif_ctx_zep = if_priv;
+	rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
+	fmac_dev_ctx = rpu_ctx_zep->rpu_ctx;
+
+	vif_ctx_zep->signal_info = si;
+	ret = wifi_nrf_fmac_get_station(rpu_ctx_zep->rpu_ctx, vif_ctx_zep->vif_idx, bssid);
+	if (ret != WIFI_NRF_STATUS_SUCCESS) {
+		LOG_ERR("%s: Failed to get station info\n", __func__);
+		goto out;
+	}
+
+	k_sem_take(&wait_for_event_sem, K_MSEC(500));
```

Comment:
the sem will be taken immediately right? IIUC, you need to initialize the count to 1 for this to block.

Reply:
NVM, I got confused, the code is right.